lib/portage/package/ebuild/fetch.py

```diff
@@ -267,6 +268,9 @@ class FlatLayout(object):
 	def get_path(self, filename):
 		return filename
 
+	def get_filenames(self, distdir):
+		return os.listdir(distdir)
```

Comment:
Let's return `iter(os.listdir(distdir))` so it returns an iterator like the FilenameHashLayout method.